src/pages/auth/register.tsx

```diff
@@ -62,11 +66,13 @@ const Register: NextPage = ({}: InferGetServerSidePropsType<
 
   const handleSignUpButtonClick = (): void => {
     // TODO: validate inputs
-    const callbackUrl = router.query.callbackUrl;
     if (callbackUrl) {
       setFormValues({ ...formValues, callbackUrl });
```

Comment:
I wonder if this guarantees `callbackUrl` will be in `formValues` 🤔. I don't think `setFormValues()` will instantly update `formValues`. Maybe we should also remove this from the `formValues` like we did with CSRF.

Reply:
Alright, updated to pass in dispatch